arroyo/dead_letter_queue/factory.py

```diff
@@ -0,0 +1,30 @@
+from typing import Callable, Mapping
+
+from arroyo.dead_letter_queue.dead_letter_queue import DeadLetterQueue
+from arroyo.dead_letter_queue.policies.abstract import DeadLetterQueuePolicy
+from arroyo.processing.strategies.abstract import (
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+)
+from arroyo.types import Partition, Position, TPayload
+
+
+class DeadLetterQueueFactory(ProcessingStrategyFactory[TPayload]):
+    """
+    DLQ Factory which wraps a given ProcessingStrategyFactory with
+    a given DLQ Policy.
+    """
```

Comment:
From what I've seen so far, we don't directly use processing strategies in Snuba. We do something like:
```python
processing_strategy_factory = SomeProcessingStrategyFactory()
processor = StreamProcessor(
    consumer,
    topic,
    processing_strategy_factory,
)

processor.run()
```

I think since the DLQ is defined as a wrapper on a processing strat/step, we can't really directly use it but instead have to pass some sort of factory to the stream processor. With this DLQ factory we can do something like this instead:

```python
processing_strategy_factory = SomeProcessingStrategyFactory()
processor = StreamProcessor(
    consumer,
    topic,
    DeadLetterQueueFactory(processing_strategy_factory, SomeDLQPolicy()),
)

processor.run()
```

Reply:
Isn't it already easy enough to incorporate the DeadLetterQueue strategy in an existing factory. i.e. just a matter of adding something like this in your `SomeProcessingStrategyFactory`
```
DeadLetterQueueFactory(strategy, SomeDLQPolicy())
```

Reply:
We could do that but I'm guessing we don't wanna change the fundamental strategy factories like `BatchProcessingStrategyFactory`, but we instead want to pick and choose which consumers we want a DLQ on?

Reply:
Is it a requirement for the strategies that use the `BatchProcessingStrategyFactory` to work with DLQ. As far as I can tell those ones never raise the `InvalidMessage` error so would never trigger the policy.

Reply:
It's not a requirement, just an example of where the DLQ could be used. Nothing raises the `InvalidMessage` exception as of now, but when the DLQ is used we'd change the relevant processor to raise the exception so the DLQ can handle stuff appropriately.

Reply:
Ok. Because I would alternatively suggest just passing it here instead and doing essentially the same thing we do with the optional filter step, e.g.  https://github.com/getsentry/arroyo/blob/main/arroyo/processing/strategies/streaming/factory.py#L126-L127.